signal-hook-registry/src/lib.rs

```diff
@@ -369,7 +480,15 @@ where
 /// Note that you really need to know what you're doing if you change eg. the `SIGSEGV` signal
 /// handler. Generally, you don't want to do that. But unlike the other functions here, this
 /// function still allows you to do it.
+#[cfg(not(windows))]
 pub unsafe fn register_unchecked<F>(signal: c_int, action: F) -> Result<SigId, Error>
+where
+    F: Fn(&siginfo_t) + Sync + Send + 'static,
+{
+    register_unchecked_(signal, action)
+}
+
+unsafe fn register_unchecked_<F>(signal: c_int, action: F) -> Result<SigId, Error>
```

Comment:
Again, I'd propose to name it something like `register_unchecked_impl` or such, not just give it a tail-underscore.